File: url_shortener/src/main/java/com/url_shortener/service/OrganizationService.java
package com.url_shortener.service;

import com.url_shortener.dto.*;
import com.url_shortener.entity.Organization;
import com.url_shortener.response.ApiResponse;
import com.url_shortener.response.OrganizationResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface OrganizationService {
    ApiResponse<OrganizationResponse> createOrganization(CreateOrganizationRequest request, String userEmail);

    ApiResponse<Page<OrganizationResponse>> getUserOrganizations(String userEmail, Pageable pageable);

    ApiResponse<OrganizationResponse> getOrganizationById(Long organizationId, String userEmail);

    ApiResponse<OrganizationResponse> updateOrganization(Long organizationId, CreateOrganizationRequest request, String userEmail);

    ApiResponse<String> deleteOrganization(Long organizationId, String userEmail);

    Organization findOrganizationEntity(Long organizationId);

    boolean hasAccess(Long organizationId, String userEmail);

    Optional<Organization> findByShortName(String shortName);
}

File: frontend/src/services/organizationService.js
import axios from "axios";
import authHeader from "./authHeader";
import config from "../config/config";

const API_URL = `${config.API_BASE_URL}${config.API_ENDPOINTS.ORGANIZATIONS}`;

const createOrganization = (data) => {
  return axios.post(API_URL, data, { headers: authHeader() });
};

const getUserOrganizations = () => {
  return axios
    .get(API_URL, {
      headers: authHeader(),
    })
    .then((response) => response.data);
};

const getOrganizationById = (id) => {
  return axios.get(`${API_URL}/${id}`, { headers: authHeader() });
};

const updateOrganization = (id, data) => {
  return axios.put(`${API_URL}/${id}`, data, { headers: authHeader() });
};

const deleteOrganization = (id) => {
  return axios.delete(`${API_URL}/${id}`, { headers: authHeader() });
};

const organizationService = {
  createOrganization,
  getUserOrganizations,
  getOrganizationById,
  updateOrganization,
  deleteOrganization,
};

export default organizationService;
